packages/smart-wallet/src/proposals/upgrade-wallet-factory2-proposal.js

```diff
@@ -56,5 +57,8 @@ export const getManifestForUpgradeWallet = (_powers, { walletRef }) => ({
       },
     },
   },
+  installations: {
+    [agoricNamesReserved.installation.walletFactory]: restoreRef(walletRef),
```

Comment:
maybe...

```suggestion
  /** @satisfies {Record<keyof agoricNamesReserved.installation, unknown>} */
  installations: {
    'walletFactory': restoreRef(walletRef),
```

Reply:
It looks good, and I'd prefer it to no check, but  it didn't enforce correct names.